src/core/scenario.jl

```diff
@@ -644,11 +647,54 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
         end
     end
 
+    # Electric Heater
     electric_heater = nothing
     if haskey(d, "ElectricHeater") && d["ElectricHeater"]["max_mmbtu_per_hour"] > 0.0
         electric_heater = ElectricHeater(;dictkeys_tosymbols(d["ElectricHeater"])...)
     end
 
+    # ASHP
+    ashp = nothing
+    cop_heating = []
+    cop_cooling = []
+    if haskey(d, "ASHP") && d["ASHP"]["max_ton"] > 0.0
```

Comment:
Depending how fast ASHP sizing solves, it may be useful to check if existing boiler and chiller is "retire_in_optimal" and then we may be able to determine the ASHP size in pre-processing, right?